internal/daemon/api_signals.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type signalsPayload struct {
+	Signal   string   `json:"signal"`
+	Services []string `json:"services"`
```

Comment:
I think we should look at adding a ProcessGroup/Group/SignalGroup bool, where if true the signal is sent to the process group of the service rather than just the pid of the process.

Reply:
That way a SIGUSR1 can just be sent to the service process and not children in the same process group.

Reply:
That's a good idea. I was wondering whether it should send to children or not. I guess there are two questions:

1) What should the default be, send to just the service, or send to group? I think your approach of sending to just the service is a reasonable default (least potentially destructive).
2) What should the field be named? I think `group` is not great, as that indicates a group name (and is used as a group name in other Pebble APIs and the plan config). Maybe `SendToGroup bool` -- longer but nice and clear?

Reply:
I'm not entirely sure if we need to signal children or not, but I'm pretty sure we should not have that as the default behavior.  We have no idea about what sub-processes the given service will have, and signaling the entire group seems somewhat of a bad practice.  The one exception being SIGKILL, probably, which we don't want to be sent this way in any case.

As such, I suggest just limiting to the process itself for now. As we learn why people want to signal the group, we can design the API and implement it accordingly.

Reply:
Okay, I've limited it to just the process for now, not the group.